src/software/backend/simulation/simulator.h

```diff
@@ -6,6 +6,51 @@
 #include "software/backend/simulation/simulator_robot.h"
 #include "software/world/world.h"
 
+extern "C"
+{
+#include "firmware/main/app/primitives/primitive_manager.h"
+}
+
+/**
+ * Because the FirmwareWorld_t struct is defined in the .c file (rather than the .h file),
+ * C++ considers it an incomplete type and is unable to use it with smart pointers
+ * because it doesn't know the size of the object. Therefore we need to create our own
+ * "Deleter" class we can provide to the smart pointers to handle that instead.
+ *
+ * See https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr for more info and
+ * examples
+ */
+struct FirmwareWorldDeleter
+{
+    void operator()(FirmwareWorld_t* firmware_world) const
+    {
+        FirmwareRobot_t* firmware_robot = app_firmware_world_getRobot(firmware_world);
+        app_firmware_robot_destroy(firmware_robot);
```

Comment:
Missing wheels on the robot. This needs to recurse down.

We should look at making the destructor for a given class actually recurse down..... but for now let's manually destroy it all here.

Reply:
fixed

Reply:
Still missing chicker/dribbler/maybe other things. Take a look at all the objects we pass via pointer to the `create` method.

Reply:
Was just missing chicker and dribbler, should be good now.